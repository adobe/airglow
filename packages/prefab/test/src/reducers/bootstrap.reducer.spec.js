/*
Copyright 2019 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

import reducer from '../../../src/reducers/bootstrap.reducer';

describe('Bootstrap.Reducer', () => {
  it('should add the specified data to the prefab store', function () {
    expect(
      reducer({}, { payload: { pizzaForm: {}, foodTable: {} } })
    ).toMatchSnapshot();
  });
  it('should merge with existing data', function () {
    const results = reducer({}, { payload: { pizzaForm: {}, foodTable: {} } });
    expect(
      reducer(results, { payload: { shakesForm: {} } })
    ).toMatchSnapshot();
  });
});
