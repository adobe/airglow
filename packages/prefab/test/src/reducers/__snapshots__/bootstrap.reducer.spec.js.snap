// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bootstrap.Reducer should add the specified data to the prefab store 1`] = `
Object {
  "foodTable": Object {
    "construct": Object {
      "name": "foodTable",
    },
    "store": Object {},
  },
  "pizzaForm": Object {
    "construct": Object {
      "name": "pizzaForm",
    },
    "store": Object {},
  },
}
`;

exports[`Bootstrap.Reducer should merge with existing data 1`] = `
Object {
  "foodTable": Object {
    "construct": Object {
      "name": "foodTable",
    },
    "store": Object {},
  },
  "pizzaForm": Object {
    "construct": Object {
      "name": "pizzaForm",
    },
    "store": Object {},
  },
  "shakesForm": Object {
    "construct": Object {
      "name": "shakesForm",
    },
    "store": Object {},
  },
}
`;
